meorg_client/cli.py

```diff
@@ -0,0 +1,250 @@
+"""Command Line Interface"""
+import click
+from meorg_client.client import Client
+import meorg_client.utilities as mcu
+import os
+import sys
+from inspect import getmembers
+import getpass
+from pathlib import Path
+import json
+
+
+def _get_client():
+    """Get an authenticated client.
+
+    Returns
+    -------
+    meorg_client.client.Client
+        Client object.
+    """
+    # Get the dev-mode flag from the environment, better than passing the dev flag everywhere.
+    dev_mode = os.getenv("MEORG_DEV_MODE", "0") == "1"
+
+    credentials = mcu.get_user_data_filepath('credentials.json')
+    credentials_dev = mcu.get_user_data_filepath('credentials-dev.json')
+
+    # In dev mode and the configuration file exists
+    if dev_mode and credentials_dev.is_file():
+        credentials = mcu.load_user_data('credentials-dev.json')
+    
+    # In dev mode and it doesn't (i.e. Actions)
+    elif dev_mode and not credentials_dev.is_file():
+        credentials = dict(
+            email=os.getenv('MEORG_EMAIL'),
+            password=os.getenv('MEORG_PASSWORD')
+        )
+    
+    # Production credentials
+    else:
+        credentials = mcu.load_user_data("credentials.json")
+
+    # Get the client
+    return Client(
+        email=credentials["email"], password=credentials["password"], dev_mode=dev_mode
+    )
+
+
+def _call(func, **kwargs):
+    """Simple wrapper to handle exceptions.
+
+    Exceptions are captured broadly and raw error message printed before non-zero exit.
+
+    Parameters
+    ----------
+    func : callable
+        Method to call.
+    **kwargs :
+        Additional arguments to method.
+
+    Returns
+    -------
+    dict
+        Response dictionary.
+    """
+    try:
+        return func(**kwargs)
+    except Exception as ex:
+        click.echo(ex.msg, err=True)
+        sys.exit(1)
+
+
+@click.group()
+def cli():
+    """
+    ModelEvaluation.org client utility.
+    """
+    pass
+
+
+@click.command()
+def list_endpoints():
+    """
+    List the available endpoints for the server.
+    """
+    client = _get_client()
+    endpoints = _call(client.list_endpoints)
+
+    for url, config in endpoints.get("paths").items():
+        for method in config.keys():
+            out = " ".join(
+                [
+                    method.upper(),
+                    url,
+                    endpoints.get("paths")[url][method]["description"],
+                ]
+            )
+
+            click.echo(out)
+
+
+@click.command()
+@click.argument("id")
+def file_status(id):
```

Comment:
Minor nitpick: does `click` support nested subcommands? I.e:

```bash
# instead of
meorg_client file_status
meorg_client file_upload
meorg_client file_list
# we have
meorg_client file status
meorg_client file upload
meorg_client file list
```

This applies also to `meorg_client analysis_*` commands.

Reply:
In a fashion, by creating multiple command groups and nesting them inside a master group we achieve this result.

It does mean that we have to disable the automated addition of commands (and we will have to remember to add every new one in code), but there are so few commands that it is probably worth it in the long run for easy of use.